### ISSUE TITLE
EventEmitter not being called by functions inheriting

### ISSUE THREAD

**Issue by fabiancook** [squaremo/amqp.node#177]
tl;dr: Why do we not set a domain on a connection

----------------------

Is there any reason not to call EventEmitter within things like Connection and Channel?

Lets take connection for example

We have this: 

`inherits(Connection, EventEmitter);`

Usually when using `inherits` you would also call the constructor

`EventEmitter.call( this )`

The reasoning for creating this issue is because if the constructor ( or `init` ) isn't called, when an event is emitted, the domain can't pick up on unhandled error events nor it can catch errors occurring in any other event. 

If there is a good reason for this not to be done then please explain, else I will make a pull request for this.
